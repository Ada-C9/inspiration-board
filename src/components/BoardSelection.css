.selection {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  text-align: center;
  margin-left: 40%;
}
