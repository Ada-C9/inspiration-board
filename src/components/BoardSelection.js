import React from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import './BoardSelection.css';

class BoardSelection extends React.Component {

	static propTypes = {
		updateBoardCallback: PropTypes.func.isRequired,
		url: PropTypes.string.isRequired,
		board: PropTypes.string
	}

	constructor(){
		super();
		this.state = {
			boards: []
		}
	}

	componentDidMount(){
		axios.get(this.props.url)
		.then((response) => {

			const updatedNames = response.data.map((obj, index) => {
				return obj.board.name
			});

			this.setState({
				boards: updatedNames
			})
		})
		.catch((error) => {

		});
	}
	onBoardUpdate = (event) => {
		console.log("in update!");
		this.props.updateBoardCallback(event.target.value);
		this.setState({
			board: event.target.value
		})
	}

	render(){
		const selectOptions = this.state.boards.map((board, index) => {
			return <option key={index} value={board}>{board}</option>
		});

		return (
			<select
				className="selection"
				onChange={this.onBoardUpdate}
				selected={this.props.board}
				>
				{selectOptions}
			</select>
		)
	}





}



export default BoardSelection;
