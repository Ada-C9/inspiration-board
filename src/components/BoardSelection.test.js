import React from 'react';
import BoardSelection from './BoardSelection';
import {shallow} from 'enzyme';

describe('BoardSelection', () => {

	test('shallow mount', () => {
		const boardSelection = shallow(<BoardSelection
			url="something"
			updateBoardCallback={() => {}}
			/>);
		expect(boardSelection).toMatchSnapshot();
	});
});
