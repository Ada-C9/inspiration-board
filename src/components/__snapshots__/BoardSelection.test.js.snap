// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSelection shallow mount 1`] = `
<select
  className="selection"
  onChange={[Function]}
/>
`;
